test/integration/integration_test.cc

```diff
@@ -1796,6 +1796,143 @@ TEST_P(IntegrationTest, Response204WithBody) {
   EXPECT_EQ(0, response->body().size());
 }
 
+// Verifies that a 304 response returns without Transfer-Encoding nor Content-Length header
+TEST_P(IntegrationTest, Response304WithoutTransferEncoding) {
```

Comment:
It would be good to also have tests that cover 304 behavior when the request is proxied over HTTP/2.  That could be done by moving these test cases to protocol_integration_test.cc

Reply:
Thank you for the pointer. I am quite new to envoy. Good to know about the protocol_integration_test. Do you think we need to do it there in this case because in terms of dealing with content body? HTTP/2 is quite different. You don't need Connection: close to indicate the body is done when no Content-Length is there (HTTP/1.0 behavior) or Transfer-Encoding: chunk (HTTP/1.1 behavior) because of the framing built into HTTP/2 protocol.

Reply:
yeah, I think the "no transfer encoding" should stay here since it's HTTP/1.1 specific, but if you could move the "handles a content length with no body" to the protocol integration test so we make sure all the versions work well, that'd be awesome.

Reply:
I have moved all the tests to protocol_integration_test.cc as the new behavior is consistent on all protocol combinations. The legacy behavior is also tested there with checks for specific upstream/downstream protocol combination.